{
  "description": "每5分钟检测一次，过去5分钟内，数据投递作业出现异常报错后，触发告警。监控目标等可在规则参数中配置。",
  "alert": {
    "name": "sls_app_export_at_error_monitor",
    "displayName": "数据投递异常报错监控",
    "type": "Alert",
    "state": "Enabled",
    "schedule": {
      "type": "FixedRate",
      "interval": "5m"
    },
    "configuration": {
      "version": "2.0",
      "type": "default",
      "dashboard": "internal-alert-analysis",
      "queryList": [
        {
          "storeType": "log",
          "project": "{{default.project}}",
          "store": "internal-diagnostic_log",
          "query": "(__topic__: etl_metrics and metric_type: ConnectorMetrics and \"_etl_:connector_metrics.state\" > 0)| select job_name, COUNT(*) as cnt, array_agg(distinct \"_etl_:connector_metrics.error\") as error  FROM log where regexp_like(\"job_name\", '.*') group by job_name limit 10000",
          "timeSpanType": "Truncated",
          "start": "-5m",
          "end": "now"
        }
      ],
      "groupConfiguration": {
        "type": "custom",
        "fields": [
          "job_name"
        ]
      },
      "joinConfigurations": [],
      "severityConfigurations": [
        {
          "severity": 8,
          "evalCondition": {
            "condition": "",
            "countCondition": ""
          }
        }
      ],
      "labels": [],
      "annotations": [
        {
          "key": "desc",
          "value": "过去5分钟内，数据投递作业(作业名称:${job_name})产生了${cnt}条报错日志reason: ${error}。更多报错细节请前往sls诊断日志库internal-diagnostic_log查看。"
        },
        {
          "key": "title",
          "value": "数据投递异常报错监控"
        },
        {
          "key": "__drill_down_query__",
          "value": "(__topic__: etl_metrics and metric_type: ConnectorMetrics and \"_etl_:connector_metrics.state\" > 0) and job_name: ${job_name}"
        }
      ],
      "autoAnnotation": true,
      "sendResolved": false,
      "threshold": 1,
      "noDataFire": false,
      "noDataSeverity": 6
    }
  }
}